File: java/selenium/src/test/java/dev/selenium/request/ModifyUrlTest.java
package dev.selenium.request;

import dev.selenium.TestBase;
import java.net.URI;
import java.util.function.Predicate;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.remote.http.HttpRequest;

public class ModifyUrlTest extends TestBase {

  @Test
  public void redirectUrl() {
      Predicate<URI> filter = uri -> uri.getHost().equals("selenium.dev");
      String url = "https://deploy-preview-2198--selenium-dev.netlify.app";

      ((FirefoxDriver) driver)
              .network()
              .addRequestHandler(filter, req -> new HttpRequest(req.getMethod(), url));

      //    UrlPattern filter = UrlPatternBuilder.setHost("selenium.dev").build();
      //    driver.network().addRequestHandler(filter, route ->
      // route.next(route.request().setUrl(url));

      driver.get("https://selenium.dev");

      String message = "Registrations Open for SeleniumConf 2025 | March 26–28 | Join Us In-Person! Register now!";
      Assertions.assertEquals(message, driver.findElement(By.tagName("h4")).getText());
  }
}

File: dotnet/Selenium/Request/ModifyUrlTest.cs
using System.Text;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.BiDi.Network;

namespace Selenium.Request;

[TestClass]
public class ModifyUrlTest : TestBase
{
    [TestMethod]
    public async Task RedirectUrl()
    {
        const string url = "https://deploy-preview-2198--selenium-dev.netlify.app";
        await using var intercept = await Bidi.Network.InterceptRequestAsync(async req =>
        {
            await req.ContinueAsync(new ContinueRequestOptions { Url = url });
        });

        // var options = new RequestInterceptOptions {
        //               UrlPatterns = new List<UrlPattern> { new UrlPattern("https://www.selenium.dev/") },
        //           };
        // await using var registration = await driver.Network.AddRequestHandlerAsync(
        //     options,
        //     req =>
        //     {
        //          req.Url = url;
        //      }
        // );


        await Driver.Navigate().GoToUrlAsync("https://selenium.dev");

        const string message =
            "Registrations Open for SeleniumConf 2025 | March 26–28 | Join Us In-Person! Register now!";
        Assert.AreEqual(message, Driver.FindElement(By.TagName("h4")).Text);
    }
}